### PR TITLE
Fix the command to build pause container

### PR DESCRIPTION
**Pull request by richardpen**
### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

This contribution is under the terms of the Apache 2.0 License:

### CODE REVIEW

File: misc/pause-container/Makefile [hunk 1]
```diff
@@ -20,7 +20,7 @@ TAG = 0.1.0
 all: container
 
 container:
-	docker build . -t "$(IMAGE):$(TAG)"
```
Comment (vsiddharth) [inline misc/pause-container/Makefile]: The build should succeed even without this patch. 
But you can change it if necessary.

Comment (richardpen) [inline misc/pause-container/Makefile]: It doesn't work on old docker version, to be specific 1.11.2. Let's make it more compatible.

Comment (vsiddharth) [inline misc/pause-container/Makefile]: Sure 👍

**Comment by richardpen** [comment]
@vsiddharth @nmeyerhans @aaithal I updated the PR with fixing the build error with `make release`, can you take another look, thanks.

File: Makefile [hunk 2]
```diff
@@ -101,6 +101,7 @@ pause-container:
 	@docker rmi -f "amazon/amazon-ecs-build-pause-bin:make"
 
 pause-container-release: pause-container
+	mkdir -p "$(shell pwd)/out"
```
Comment (nmeyerhans) [inline Makefile]: `$(shell pwd)/` is redundant. You're referencing it relative to the working directory in the next line, so you might as well do it here.

File: Makefile [hunk 3]
```diff
@@ -101,7 +101,8 @@ pause-container:
 	@docker rmi -f "amazon/amazon-ecs-build-pause-bin:make"
 
 pause-container-release: pause-container
-	@docker save ${PAUSE_CONTAINER_IMAGE}:${PAUSE_CONTAINER_TAG} > out/${PAUSE_CONTAINER_TARBALL}
+	mkdir -p "$(shell pwd)/out"
+	@docker save ${PAUSE_CONTAINER_IMAGE}:${PAUSE_CONTAINER_TAG} > "$(shell pwd)/out/${PAUSE_CONTAINER_TARBALL}"
```
Comment (nmeyerhans) [inline Makefile]: `$(shell pwd)/` is still redundant, but it doesn't impact functionality.

:ship:

**Pull request merged**